### PR TITLE
[SCP-1496] feat: modify product factory

### PR DESCRIPTION
**Pull request by miguel-serfe**
Coverage:

gsc-master:
```
  Classes: 93.28% (111/119)  
  Methods: 97.77% (525/537)  
  Lines:   93.77% (3083/3288)
```
feat/SCP-1496-modify-product-factory:
```
  Classes: 93.28% (111/119)  
  Methods: 97.78% (528/540)  
  Lines:   93.81% (3107/3312)
```

### CODE REVIEW

File: src/Factory/Xml/Product/ProductFactory.php [hunk 1]
```diff
@@ -137,4 +116,100 @@ public static function make(SimpleXMLElement $element): Product
 
         return $product;
     }
+
+    public static function makeGlobalProduct(SimpleXMLElement $element): GlobalProduct
```
Comment (sebastian-serfe) [inline src/Factory/Xml/Product/ProductFactory.php:120]: should this method be privated?

File: src/Factory/Xml/Product/ProductFactory.php [hunk 2]
```diff
@@ -137,4 +116,100 @@ public static function make(SimpleXMLElement $element): Product
...
+
+        if (!empty($element->MainImage)) {
+            $image = new Image((string) $element->MainImage);
+            $product->setMainImage($image);
+        }
+
+        return $product;
+    }
+
+    private static function ValidateBaseProductXmlStructure(SimpleXMLElement $element): void
```
Comment (sebastian-serfe) [inline src/Factory/Xml/Product/ProductFactory.php:177]: Make a correct use of camel case naming convention

File: tests/Unit/Product/GlobalProductTest.php [hunk 3]
```diff
@@ -206,4 +274,99 @@ public function testItReturnsAJsonRepresentation(): void
...
         $this->assertJsonStringEqualsJsonString(Json::encode($expectedJson), Json::encode($product));
     }
+
+    /**
+     * @dataProvider invalidXmlStructure
+     */
+    public function testItThrowsAExceptionWithoutASellerSkuInTheXml($property): void
+    {
+        $xmlString = sprintf(
+            $this->getSchema('Product/GlobalProduct.xml'),
```
Comment (sebastian-serfe) [inline tests/Unit/Product/GlobalProductTest.php:284]: The xml string creation here is the same than the assignment on testItMakesAProductFromXml, can you create a method to assign and return that?

**Review by sebastian-serfe — CHANGES_REQUESTED** [review]

File: src/Factory/Xml/Product/ProductFactory.php (hunk 1 above)
Comment (miguel-serfe) [reply src/Factory/Xml/Product/ProductFactory.php]: Yes, it should be. Thanks!

File: tests/Unit/Product/GlobalProductTest.php (hunk 3 above)
Comment (miguel-serfe) [reply tests/Unit/Product/GlobalProductTest.php]: I've refactored this file and the product test with the same change. Thanks!

File: src/Factory/Xml/Product/ProductFactory.php [hunk 4]
```diff
@@ -137,4 +116,100 @@ public static function make(SimpleXMLElement $element): Product
...
+    {
+        if (!property_exists($element, 'SellerSku')) {
```
[the remark this answers is not in the document]
Comment (miguel-serfe) [reply src/Factory/Xml/Product/ProductFactory.php]: Done! thanks!

**Review by serfe-linio-francop — APPROVED** [review]

**Review by sebastian-serfe — APPROVED** [review]
well done dude!

**Review by maruamallo — APPROVED** [review]

**Review by yinosanchez — APPROVED** [review]

**Pull request merged**